core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSender42.java

```diff
@@ -66,16 +73,26 @@ public void dispose(HttpResponse response) {
 
     @Override
     public void close() {
-        try {
-            cm.shutdown();
-        } catch (Exception e) {
-            // chomp
+        synchronized (lock) {
+            if (cm != null) {
+                cm.shutdown();
```

Comment:
This may not be necessary since `HttpClientBuilder.setConnectionManagerShared` is not set, i.e. the connection manager to http client relationship is 1-1 and the client should shutdown the CM when it's done.